wsi/drm/drmbuffer.cpp

```diff
@@ -77,23 +77,43 @@ DrmBuffer::~DrmBuffer() {
 #endif
 }
 
+void DrmBuffer::SaveMetaData(const HwcBuffer& bo) {
+  image_.handle_->meta_data_.width_ = bo.width_;
+  image_.handle_->meta_data_.height_ = bo.height_;
+  image_.handle_->meta_data_.format_ = bo.format_;
+  image_.handle_->meta_data_.tiling_mode_ = bo.tiling_mode_;
+  image_.handle_->meta_data_.native_format_ = bo.native_format_;
+  image_.handle_->meta_data_.num_planes_ = bo.num_planes_;
+  image_.handle_->meta_data_.usage_ = bo.usage_; /* defaulted: kLayerNormal */
+
+  for (uint32_t i = 0; i < 4; i++) {
+    image_.handle_->meta_data_.pitches_[i] = bo.pitches_[i];
+    image_.handle_->meta_data_.offsets_[i] = bo.offsets_[i];
+    image_.handle_->meta_data_.gem_handles_[i] = bo.gem_handles_[i];
+    image_.handle_->meta_data_.prime_fds_[i] = bo.prime_fds_[i];
+  }
+
+  for (uint32_t i = 0; i < 8; i++) {
+    image_.handle_->meta_data_.fb_modifiers_[i] = bo.fb_modifiers_[i];
+  }
+}
+
 void DrmBuffer::Initialize(const HwcBuffer& bo) {
   width_ = bo.width_;
   height_ = bo.height_;
-  for (uint32_t i = 0; i < 4; i++) {
-    pitches_[i] = bo.pitches_[i];
-    offsets_[i] = bo.offsets_[i];
-    gem_handles_[i] = bo.gem_handles_[i];
-  }
 
-  format_ = bo.format_;
+  SaveMetaData(bo);
+  width_ = GetWidth();
+  height_ = GetHeight();
+
+  format_ = GetFormat();
```

Comment:
I think we talked about this off-line, but I can't remember the reasoning behind keeping some of these copies around. Would it be possible to drop these and use only the image_.handle_->meta_data_ copies?